### PR TITLE
feat: allow installer to download releases for ARM64

### PR DESCRIPTION
**Pull request by kelvintaywl**
# Checklist

=========

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked for similar issues and haven't found anything relevant.
- [x] This is not a security issue (which should be reported here: https://circleci.com/security/)
- [x] I have read [Contribution Guidelines](https://github.com/CircleCI-Public/circleci-cli/blob/master/CONTRIBUTING.md).

### Internal Checklist
- [ ] I am requesting a review from my own team as well as the owning team
- [ ] I have a plan in place for the monitoring of the changes that I am making (this can include new monitors, logs to be aware of, etc...)

## Changes

=======

- Update _install.sh_ to account for CPU architecture when downloading the CLI binary releases

## Rationale

A customer noted that running the script [according to our docs here](https://circleci.com/docs/local-cli/#alternative-installation-method), would install the CircleCI CLI for linux/amd64 even if the machine is linux/arm64.

I have also tested these changes on a public sample repo with this modified _install.sh_ here:
https://github.com/kelvintaywl-cci/circleci-cli-install ([commit](https://github.com/kelvintaywl-cci/circleci-cli-install/commit/abe0dd7c8396c9f90e05ca1016977321219261c8))

We can see the builds for MacOS and Linux (arm64 + amd64):
https://app.circleci.com/pipelines/github/kelvintaywl-cci/circleci-cli-install/3

### CODE REVIEW

**Review by corinnesollows — COMMENTED** [review]
Our Code yellow review process is now:
1. Review from own team
2. Review from someone on another team
3. EM and director approval in the Jira ticket
4. Post in [#engineering](https://circleci.slack.com/archives/C07M73TT8) to get a review from a final approver
Extensibility would be your #2. Your code looks good to me, but I will hit approve when I see more reviewers pending. :)

**Review by rlegan — APPROVED** [review]

**Comment by tgriffin-ls** [comment]
I tried this on AlmaLinux 9.1 (arm64) and the installation itself works fine, but there is an issue with the verification at the end:
```
command -v circleci
```
On AlmaLinux (and everything in the Fedora/RHEL family) `/usr/local/bin` is not in root-users `PATH`. So this verification fails.

**Comment by kelvintaywl** [comment]
hey team @CircleCI-Public/support 

I am hoping if anyone can help review this PR 🙇 
thank you!

I've updated my verification workflow here, and can confirm this PR's install.sh worked for Linux arm64, amd64 and Mac (amd64):
https://app.circleci.com/pipelines/github/kelvintaywl-cci/circleci-cli-install/6/workflows/b732820f-19fd-449a-8bc8-6dca92eac501

**Comment by nanophate** [comment]
I was able to also check this working on following machine. LGTM 👍 ✅ 

```
naoya@instance-1:~$ uname -a
Linux instance-1 6.1.0-0.deb11.6-cloud-arm64 #1 SMP Debian 6.1.15-1~bpo11+1 (2023-03-16) aarch64 GNU/Linux
naoya@instance-1:~$ curl -fLSs https://raw.githubusercontent.com/kelvintaywl-cci/circleci-cli/update-install.sh/install.sh | sudo bash
Starting installation.
Installing CircleCI CLI v0.1.26343
Installing to /usr/local/bin
/usr/local/bin/circleci
```

**Comment by JulesFaucherre** [comment]
@tgriffin-ls , thanks for reporting! The issue you're talking about is not really related to the one that is fixed in this pr, but we're creating a ticket and we'll handle this

**Pull request merged**
